psi4/src/psi4/libmints/mintshelper.cc

```diff
@@ -619,9 +618,25 @@ SharedMatrix MintsHelper::ao_dkh(int dkh_order)
     H_dk->subtract(Torig);
 
     H_dk->set_name("AO-basis DKH Ints");
-    //H_dk->print();
+//    H_dk->print();
+
+    // DKH is solved in the decontracted basis ... transform to the contracted.
+
+    SharedMatrix S_inv = S->clone();
+    S_inv->general_invert();
```

Comment:
there's an SVD-based pseudoinverse in Matrix that could be used for singular cases